src/Fields/interpolate.jl

```diff
@@ -88,12 +81,32 @@ end
     return fractional_index(x, xn, Nx) 
 end
 
-@inline function fractional_x_index(x, locs, grid::LatitudeLongitudeGrid)
+@inline convert_to_0_360(x) = ((x % 360) + 360) % 360
+
+@inline convert_to_minus_180_180(x) = rem(x, 360, RoundNearest)
+
+# When interpolating longitude values, we convert all longitudes to the 0-360 range
+# if the parent grid starts with a positive number (λ₀ > 0), and to a -180-180 range
+# if the parent grid starts with a negative number (λ₀ < 0).
```

Comment:
I don't get it. Is this a generic solution?

Reply:
I'm not sure it makes sense to define valid lat-lon limits here where we do interpolation. It may make more sense to enforce this in the definition of the grids, if we are trying to do this. Or somehow encode information about the coordinates being used in the grid rather than implicitly infer it here.

Reply:
This is quite generic but not 100%, it does not allow grids that have (for example) a range of (-x : 360-x) if x < 180. I have to think about how we can convert to the range `λ₀ : λ₀+360` that would be completely general and allow interpolating from any lat - lon grid to any `λ` value.

Reply:
enforcing it in the grid might be very complicated because the longitude coordinate should be monotone if we want to support interpolation.

Reply:
> This is quite generic but not 100%, it does not allow all type of grids defined on the range (-x : 360-x). I have to think about how we can convert to the range `λ₀ : λ₀+360` that would be completely general and allow interpolating from any lat - lon grid to any `λ` value.

I think this is pretty easy to enforce, so I will be changing this PR to do it